Source/JavaScriptCore/jit/AssemblyHelpers.cpp

```diff
@@ -332,7 +332,9 @@ void AssemblyHelpers::emitStoreStructureWithTypeInfo(AssemblyHelpers& jit, Trust
 {
     const Structure* structurePtr = reinterpret_cast<const Structure*>(structure.m_value);
 #if USE(JSVALUE64)
-    jit.store64(TrustedImm64(structurePtr->idBlob()), MacroAssembler::Address(dest, JSCell::structureIDOffset()));
+    uint64_t id = static_cast<uint64_t>(StructureID::encode(structurePtr).bits());
+    uint64_t blob = static_cast<uint64_t>(structurePtr->idBlob());
+    jit.store64(TrustedImm64(id | (blob << 32)), MacroAssembler::Address(dest, JSCell::structureIDOffset()));
```

Comment:
Is this really the correct order? I would have expected id to be the upper bits

Reply:
Yes, the structure id is in the lower bits and the other info data are in higher bits.